lib/aerospike/node/refresh/friends.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module Aerospike
+  class Node
+    module Refresh
+      # Refresh peers/friends based on old service protocol
+      module Friends
+        class << self
+          def call(node, peers, info_map)
+            friend_string = info_map['services']
+            cluster = node.cluster
+
+            if friend_string.to_s.empty?
+              node.peers_count.value = 0
+              return
+            end
+
+            friend_names = friend_string.split(';')
+            node.peers_count.value = friend_names.size
+
+            friend_names.each do |friend|
+              hostname, port = friend.split(':')
+              host = Host.new(hostname, port.to_i)
+              found_node = cluster.find_alias(host)
+
+              if found_node
+                found_node.increase_reference_count!
+              else
+                unless peers.hosts.include?(host)
+                  prepare(cluster, peers, host)
+                end
+              end
+            end
+          end
+
+          def prepare(cluster, peers, host)
```

Comment:
You still have some duplicate code (merge artifacts?) in this method. You should not be using `Cluster::FindNode` here, which should be used only when using "peers" for cluster discovery. Please refer to the Java client's [`prepareFriend`](https://github.com/aerospike/aerospike-client-java/blob/e82206496cb3d33222315fb978f7c21ef63251d1/client/src/com/aerospike/client/cluster/Node.java#L263) method instead.

This code is only used when connecting to older servers that do not support the peers protocol. (v3.9 and older, I believe.)

Reply:
Fixed this. However, I'm still seeing nodes being "added" on every refresh cycle when running eg. a three node cluster. Debugging why, but my nodes have two hostnames (two IPs) each so I suspect that has something to do with it